controller/Lists.php

```diff
@@ -21,54 +21,47 @@
  *               2013-2021 (update and modification) Open Assessment Technologies SA;
  */
 
+declare(strict_types=1);
+
 namespace oat\taoBackOffice\controller;
 
+use tao_helpers_Uri;
+use RuntimeException;
+use common_Exception;
+use common_exception_Error;
+use oat\tao\helpers\Template;
+use tao_actions_CommonModule;
+use tao_helpers_Scriptloader;
+use core_kernel_classes_Class;
 use common_exception_BadRequest;
-use oat\taoBackOffice\model\lists\ListCreatedResponse;
-use common_ext_ExtensionException as ExtensionException;
-use core_kernel_classes_Class as RdfClass;
-use core_kernel_classes_Property as RdfProperty;
+use tao_actions_form_RemoteList;
+use common_ext_ExtensionException;
+use oat\generis\model\data\Ontology;
 use core_kernel_persistence_Exception;
-use oat\generis\model\OntologyAwareTrait;
-use oat\tao\helpers\Template;
-use oat\tao\model\featureFlag\FeatureFlagChecker;
 use oat\tao\model\http\HttpJsonResponseTrait;
-use oat\tao\model\Lists\Business\Domain\CollectionType;
 use oat\tao\model\Lists\Business\Domain\Value;
+use oat\taoBackOffice\model\lists\ListCreator;
+use oat\taoBackOffice\model\lists\ListService;
+use oat\tao\model\featureFlag\FeatureFlagChecker;
+use oat\taoBackOffice\model\lists\ListCreatedResponse;
+use oat\tao\model\Lists\Business\Service\RemoteSource;
+use oat\tao\model\Lists\Business\Domain\CollectionType;
 use oat\tao\model\Lists\Business\Domain\ValueCollection;
 use oat\tao\model\featureFlag\FeatureFlagCheckerInterface;
 use oat\tao\model\Lists\Business\Domain\RemoteSourceContext;
-use oat\tao\model\Lists\Business\Domain\ValueCollectionSearchRequest;
+use oat\tao\model\Lists\Business\Service\ValueCollectionService;
 use oat\tao\model\Lists\Business\Input\ValueCollectionSearchInput;
-use oat\tao\model\Lists\Business\Service\RemoteSource;
 use oat\tao\model\Lists\Business\Service\RemoteSourcedListOntology;
-use oat\tao\model\Lists\Business\Service\ValueCollectionService;
+use oat\taoBackOffice\model\ListElement\Service\ListElementsFinder;
+use oat\tao\model\Lists\Business\Domain\ValueCollectionSearchRequest;
 use oat\tao\model\Lists\DataAccess\Repository\ValueConflictException;
-use oat\tao\model\TaoOntology;
-use oat\taoBackOffice\model\lists\ListCreator;
-use oat\taoBackOffice\model\lists\ListService;
-use RuntimeException;
-use tao_actions_CommonModule;
-use tao_actions_form_RemoteList;
-use tao_helpers_Scriptloader;
-use tao_helpers_Uri;
-use tao_models_classes_LanguageService;
+use oat\taoBackOffice\model\ListElement\Context\ListElementsFinderContext;
+use oat\taoBackOffice\model\ListElement\Contract\ListElementsFinderInterface;
 
-/**
- * This controller provide the actions to manage the lists of data
- *
- * @author CRP Henri Tudor - TAO Team - {@link http://www.tao.lu}
- * @license GPLv2  http://www.opensource.org/licenses/gpl-2.0.php
- * @package taoBackOffice
- *
- */
 class Lists extends tao_actions_CommonModule
 {
```

Comment:
This controller seems super big (near 600 lines), I'd try to move some logic to the model is possible.

Reply:
Yeah, I agree on this point. This is also a part of my tasks.
But I'm not sure, that we have time to do this during this task (this one is sprint goal), so I will not do this in during this task/PR

Reply:
That's fine, just saying because I found a bit difficult to follow all changes here :sweat_smile: 

BTW I'll close this thread :+1: